xbmc/video/VideoDatabase.cpp

```diff
@@ -4028,30 +4096,19 @@ std::string CVideoDatabase::GetFileBasePathById(int idFile)
 
 int CVideoDatabase::GetFileIdByMovie(int idMovie)
 {
-  if (!m_pDB || !m_pDS)
-    return -1;
-
-  int idFile = -1;
-
-  try
-  {
-    m_pDS->query(PrepareSQL("SELECT idFile FROM movie WHERE idMovie = %i", idMovie));
-
-    if (!m_pDS->eof())
-    {
-      idFile = m_pDS->fv("idFile").get_asInt();
-    }
-
-    m_pDS->close();
+  const std::string query{PrepareSQL("SELECT idFile FROM movie WHERE idMovie = %i", idMovie)};
+  const int idFile{GetSingleValueInt(query, m_pDS)};
+  return idFile > 0 ? idFile : -1;
```

Comment:
It's OK to rely more on helper functions but this cosmetic change is not related to the fix and doesn't belong to this commit (or PR).